R/plotScoresByCellOrLabel.R

```diff
@@ -0,0 +1,128 @@
+    }
+    df <- plotScores.dataGather(results, prune.calls, labels.use)
+    p <- ggplot(data = df[df$id == rownames(results)[cell.id],],
+                aes(x = label, y = score, fill = called.this)) + 
+        theme_classic() + geom_point(
+                color = "black",
+                shape = 21,
+                size = size,
+                alpha = 1) + scale_fill_manual(values = colors) +
+        theme(axis.text.x= element_text(angle=60, hjust = 1, vjust = 1, size=12)) +
+        xlab(NULL)
+    p
+}
+
+#' @describeIn plotScoresByCellOrLabel Plot scores accross labels of an individual cells
+#' @export
+plotScoresSingleLabel <- function(results, prune.calls = NULL, label, size = 0.5, dots.on.top = FALSE, df = NULL,
+    colors = c("#F0E442", "#56B4E9", "gray70", "gray40")){
+    if(length(colors)<4){stop("4 colors are expected.")}
+    if (is.null(names(colors))){names(colors) <- c('this label', 'this label - pruned', 'other label', 'other label - pruned')}
+    if (is.null(rownames(results))) {
+        rownames(results) <- seq_len(nrow(results))
+    }
+    if (is.null(df)) {df <- plotScores.dataGather(results, prune.calls, label)}
+    p <- ggplot(data = df[df$label == label,],
+                aes(x = called.this, y = score, fill = called.this)) + 
+        theme_classic() +
+        scale_fill_manual(values = colors) + 
+        scale_x_discrete(name = label, labels = NULL)
+    if(dots.on.top){ p <- p+ geom_violin()}
+    p <- p + geom_jitter(height = 0, width = 0.3, color = "black", shape = 16, size = size)
+    if(!dots.on.top){ p <- p + geom_violin()}
+    p
+}
+
+#' @describeIn plotScoresByCellOrLabel Plot scores accross labels of an individual cells
+#' @export
+plotScoresMultiLabels <- function(results, prune.calls = NULL, size = 0.2, dots.on.top = FALSE,
+    labels.use = levels(as.factor(results$labels)), ncol = 5,
+    colors = c("#F0E442", "#56B4E9", "gray70", "gray40"), ...){
+    if(length(colors)<4){stop("4 colors are expected.")}
+    if (is.null(names(colors))){names(colors) <- c('this label', 'this label - pruned', 'other label', 'other label - pruned')}
+    if (is.null(rownames(results))) {
+        rownames(results) <- seq_len(nrow(results))
+    }
+    df <- plotScores.dataGather(results, prune.calls, labels.use)
+    max <- max(df$score)
+    plots <- lapply(labels.use, function(X) {
+        plotScoresSingleLabel(results, prune.calls, label = X, size, dots.on.top, df = df, colors) +
+            theme(legend.position = "none", axis.ticks.x=element_blank()) +
+            coord_cartesian(ylim = c(0,max)) + ylab(NULL)
+    })
+    plots <- c(plots,
+               list(cowplot::ggdraw(cowplot::get_legend(
+        plotScoresSingleLabel(results, prune.calls = prune.calls, label = labels.use[1])))))
+    gridExtra::grid.arrange(grobs=plots, ncol = ncol, ...)
+}
+
+plotScores.dataGather <- function(results, prune.calls = NULL, labels.use = levels(as.factor(results$labels)))
```

Comment:
Internal functions tend to look like `.blah_stuff()`; note the dot prefix and the underscore separator. This makes it easy to distinguish internal from non-internal function sat a glance.